### ISSUE TITLE
400 error on all 3rd-party requests

### ISSUE THREAD

**Issue by hesedel** [stereobooster/react-snap#217]
Why do I get 400 errors during running react snap on 3rd-party requests? When loading through browser, everything is fine. But during react-snap's process they are all logged as 400 errors and therefore the static pages are saved without them.

**Comment by stereobooster** [comment]
> 400 Bad Request 

Do you have access to server logs? Can you open exactly same requests in the browser?

**Comment by hesedel** [comment]
So, this is what I'm getting on the server:

`c3828630a46dad2eebf0353dac41d1e84f144649869cc77aa138a2ffc978ad18 escada-i18n [21/Jul/2018:12:03:22 +0000] 35.237.252.116 - F0D4106598B5F4A3 WEBSITE.GET.OBJECT 198577/en.json "GET /198577/en.json HTTP/1.1" 200 - 46 46 19 18 "http://localhost:45678/404" "ReactSnap" `

Is that `200` - OK?

but on the react-snap logs i'm getting this:
<img width="636" alt="screen shot 2018-07-21 at 15 41 12" src="https://user-images.githubusercontent.com/1061404/43036187-c8b5b756-8cfc-11e8-9199-154b7943fa0c.png">

The request data is also not being set on window.snapStore

**Comment by hesedel** [comment]
UPDATE: I'm doing console logs on the fetch success and seeing them on build, it's actually fetching the data okay! the 400 error seems to be when repeating the request on other pages.

The my only problem now is that the result is not being saved statically or saved to snapStore like the other local requests.

**Comment by stereobooster** [comment]
```js
  page.on("response", response => {
    if (response.status() >= 400) {
      console.log(`⚠️   ${response.status()} error: ${response.url()}`);
    }
  });
```

so this is not critical error, this is just info for you, so that this request is resulted in error. Can you open the request in the incognito mode window?

**Comment by hesedel** [comment]
Yes, I'm always developing in incognito window. All is working fine. It seems react snap is also getting the response fine, that's why it's weird to get the 400 error in the log. Is it also trying to re-request? As it looks like some requests repeat. I'm attaching screenshots to show what I mean:

<img width="931" alt="screen shot 2018-07-21 at 19 15 18" src="https://user-images.githubusercontent.com/1061404/43038605-acb8a3e0-8d1c-11e8-9b19-c8b427c3de7b.png">
<img width="949" alt="screen shot 2018-07-21 at 19 16 51" src="https://user-images.githubusercontent.com/1061404/43038606-acd17ca8-8d1c-11e8-8277-b04ff5c4aadc.png">

**Comment by stereobooster** [comment]
The issue here is not in your app, not in react-snap, not in the headless browser, the issue in the network. You will need network debugging proxy to understand why those requests fail, maybe it is your firewall.

**Comment by hesedel** [comment]
Okay, could be, but it's also the same errors I see when it runs in GitLab CI, so it's two different systems.

**Comment by stereobooster** [comment]
Hm.. not firewall okay. Second hypothesis - they changed something in headless chrome. What version of chrome, puppeteer and OS?

**Comment by hesedel** [comment]
chrome: Version 67.0.3396.99 (Official Build) (64-bit)
puppeteer: 1.6.0
Dev OS: macOS High Sierra 10.13.6
GitLab OS: some linux instance

**Comment by stereobooster** [comment]
Well I actually able to reproduce it. In my case it is caused by `"inlineCss": true`

**Comment by hesedel** [comment]
With that I guess it's because with inlineCss, you have double request with the inline+external loading.

**Comment by stereobooster** [comment]
There is no real reason to disallow cache external requests. I guess I never need it this is why

**Comment by hesedel** [comment]
just made a small PR, just to include external json requests to the cache. #220
